operator/pkg/helmreconciler/reconciler.go

```diff
@@ -527,6 +528,18 @@ func (h *HelmReconciler) analyzeWebhooks(whs []string) error {
 		return nil
 	}
 
+	exists := revtag.PreviousInstallExists(context.Background(), h.kubeClient.Kube())
```

Comment:
I think the test is failed since `PreviousInstallExists ` and `GenerateTagWebhookYAML ` are trying to do some webhook list request.

There's some comments here:
https://github.com/istio/istio/blob/09917748493f1e29389d6c5eb9dee0f99db796fc/pkg/kube/client.go#L238-L245

In tests, when having a list request there will be `c.informerWatchesPending.Inc()`, but it will never dec(). As a result it's never marked as synced:
```
2023-05-26T07:35:26.259446Z	info	installer	Creating Service/istio-operator/istio-operator (istiod/)

✔ Istio operator installed2023-05-26T07:35:26.270014Z	info	installer	Creating IstioOperator/istio-system/istiod (istiod/)
2023-05-26T07:35:26.866452Z	info	cluster "" kube client started
2023-05-26T07:35:31.389130Z	info	waiting for sync...	name=crd watcher attempt=50 time=4.52242975s
2023-05-26T07:35:36.452707Z	info	waiting for sync...	name=crd watcher attempt=100 time=9.58613525s
2023-05-26T07:35:41.512890Z	info	waiting for sync...	name=crd watcher attempt=150 time=14.64648475s
2023-05-26T07:35:46.569947Z	info	waiting for sync...	name=crd watcher attempt=200 time=19.703555917s
```

@howardjohn Do you have any suggestions?

Reply:
I've tried to add timeout but it made the test duration > 1min